pkgs/applications/graphics/vengi-tools/default.nix

```diff
@@ -92,19 +100,19 @@ stdenv.mkDerivation rec {
     run-voxedit = nixosTests.vengi-tools;
   };
 
-  meta = with lib; {
+  meta = {
     description = "Tools from the vengi voxel engine, including a thumbnailer, a converter, and the VoxEdit voxel editor";
+    downloadPage = "https://github.com/mgerhardy/vengi/releases";
+    homepage = "https://mgerhardy.github.io/vengi/";
+    license = with lib.licenses; [ mit cc-by-sa-30 ];
     longDescription = ''
       Tools from the vengi C++ voxel game engine. It includes a voxel editor
       with character animation support and loading/saving into a lot of voxel
       volume formats. There are other tools like e.g. a thumbnailer for your
       filemanager and a command line tool to convert between several voxel
       formats.
     '';
-    homepage = "https://mgerhardy.github.io/vengi/";
-    downloadPage = "https://github.com/mgerhardy/vengi/releases";
-    license = [ licenses.mit licenses.cc-by-sa-30 ];
-    maintainers = with maintainers; [ fgaz ];
-    platforms = platforms.all;
+    maintainers = with lib.maintainers; [ fgaz ];
+    platforms = lib.platforms.all;
```

Comment:
Is `with lib` discouraged in `meta` now? Last I checked `meta` was one of the few places where it's appropriate.
I also think we should avoid gratuitous reformatting such as sorting meta elements, unless we're already changing them.

Reply:
I understand.

Regarding the `with lib;`:

There are several compelling reasons to avoid using a `with` statement.

Firstly, employing such a statement imports all the functions from `lib` into the current scope. This can potentially cloud your code's clarity and lead to confusion. A more preferable alternative would be using a targeted statement like `inherit (lib) abc def;` which enhances the readability of your code. This way, you can accurately trace the origin of each function.

Secondly, a `with` statement could interfere with your Language Server Protocol (LSP) and break static analysis at some point. Given that the entire `lib` scope is imported, your LSP might struggle to pinpoint the sources of various functions, resulting in a poor development experience.

There are additional considerations regarding this topic that fall outside my expertise, but I encourage you to post this question on discourse for a more comprehensive discussion.

You can find a very good article about that on [nix.dev](https://nix.dev/recipes/best-practices#with-scopes).

There is also this RFC: https://github.com/r-burns/nixos-rfcs/blob/rfc-inherit-as-list/rfcs/0110-inherit-as-list.md

Regarding the attributes sorting:

Having attributes sorted facilitate finding them when looking for them.
Also, this is usually what I recommend in all the PRs that I'm reviewing.
And this is mimicking the behaviour of the nix language by default.
Obviously these changes were not mandatory, but since I was reformatting the file, I took the liberty to sort them properly within the commit.


Look, I'm not here to convince anyone or anybody, if you don't like the changes, just let me know, and I'll revert immediately, and add a note in my list that I should not edit this package any more in the future.

Reply:
> Regarding the `with lib;`: [...]

Yes I understand that `with` is generally bad and I agree, that's why I asked specifically about `meta` (which afaik is the only place where `with` is widely accepted common practice), but your answer is about general use.

> Look, I usually do these small refactoring in PRs made by r-ryantm and only when it's a bug fixing release. I'm trying to improve the quality of `nixpkgs` piece by piece, one at a time when I have some free time. I'm also quite pragmatic, and I'm not here to convince anyone or anybody. If you don't like the changes or agree for any reason, just let me know, and I'll remove the commit immediately, and add a note in my list that I should not edit this package any more in the future.

Yes but keep the `finalAttrs` and `makeBinaryWrapper` changes, those are good.
Also I'd appreciate if you added all the packages I maintain to your list. I'd like to avoid reformats.

Please don't take this personally, we're all trying to improve nixpkgs here :)

Reply:
Understood! Commit removed.

Reply:
Thanks!